org.eclipse.m2e.jdt/src/org/eclipse/m2e/jdt/internal/AbstractJavaProjectConfigurator.java

```diff
@@ -496,9 +512,14 @@ private void addResourceDirs(IClasspathDescriptor classpath, IProject project, M
         continue;
       }
       File resourceDirectory = new File(directory);
-      IPath relativePath = getProjectRelativePath(project, directory);
-      IResource r = project.findMember(relativePath);
-      if(r == project) {
```

Comment:
This check does not work anymore, since r is now always an instance/handle for an IFolder and cannot be a project anymore.
Unfortunately my alternative I just have pushed doesn't work either since the absolute path is resolved against the project (with stripped-off drive), when `project.getFolder(projectLocation)` is called.

I have to think about another elegant/efficient solution to check if `resourceCanonicalPath` points to the project location.

Reply:
I didn't found a solution as simple/fast as the identity check before, so went on with the obvious one and changed it to compare the absolute resource path with the absolute project location.

In general this class could benefit from a medium clean-up where the re-compuation of the project-location could be avoided. But for now this is fine.